yarpl/include/yarpl/Refcounted.h

```diff
@@ -176,35 +181,124 @@ bool operator==(const Reference<T>& lhs, const Reference<U>& rhs) noexcept {
   return lhs.get() == rhs.get();
 }
 
+template <typename T>
+bool operator==(const Reference<T>& lhs, std::nullptr_t) noexcept {
```

Comment:
These lines are added for optimization purposes. std::shared_ptr also has the same lines. It saves the compiler from generating an instance of Reference<nullptr_t>. Otherwise, for each such check
```if (variable == nullptr)```
we would be creating an instance of Reference<nullptr_t>.